src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -1413,6 +1420,8 @@ internal partial class BoundDynamicIndexerAccess
     {
         protected override OperationKind ExpressionKind => OperationKind.None;
 
+        protected override ImmutableArray<IOperation> Children => this.Arguments.Add(this.ReceiverOpt).As<IOperation>();
```

Comment:
>  I felt it is better to add all nodes and the client handle null cases.

Then we should clearly document this on the API in the BoundNode and in the interface.